### PR TITLE
BAH-1971 | Getting requester information on the reports table

### PR DESCRIPTION
**Pull request by Arjun-Go**
JIRA link - https://bahmni.atlassian.net/jira/software/c/projects/BAH/boards/35?modal=detail&selectedIssue=BAH-1971

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/fhirExtension/translators/impl/ObsBasedDiagnosticReportTranslatorImpl.java [hunk 1]
```diff
@@ -32,16 +38,23 @@ public class ObsBasedDiagnosticReportTranslatorImpl implements ObsBasedDiagnosti
...
+	@Autowired
+	private ProviderService providerService;
+	
+	static final String PRACTITIONER = "Practitioner";
+	
 	@Override
 	public DiagnosticReport toFhirResource(@Nonnull FhirDiagnosticReport fhirDiagnosticReport) {
 		DiagnosticReport diagnosticReport = diagnosticReportTranslator.toFhirResource(fhirDiagnosticReport);
 		setPresentedForm(diagnosticReport, fhirDiagnosticReport);
+		setPractitioner(diagnosticReport, fhirDiagnosticReport);
```
Comment (pajmera4) [inline api/src/main/java/org/openmrs/module/fhirExtension/translators/impl/ObsBasedDiagnosticReportTranslatorImpl.java:50]: There is already one translator in fhir2 codebase: PractitionerReferenceTranslatorProviderImpl.
Can we reuse the same for the purpose.

File: api/src/main/java/org/openmrs/module/fhirExtension/translators/impl/ObsBasedDiagnosticReportTranslatorImpl.java [hunk 2]
```diff
@@ -41,7 +41,8 @@ public class ObsBasedDiagnosticReportTranslatorImpl implements ObsBasedDiagnosti
 	@Autowired
 	private ProviderService providerService;
 	
-	static final String PRACTITIONER = "Practitioner";
+	@Autowired
+	private PractitionerReferenceTranslator<Provider> practitionerReferenceTranslator;
```
Comment (pajmera4) [inline api/src/main/java/org/openmrs/module/fhirExtension/translators/impl/ObsBasedDiagnosticReportTranslatorImpl.java:45]: practitionerReferenceTranslator also need to be called from toOpenMrsType function.

File: api/src/main/java/org/openmrs/module/fhirExtension/translators/impl/ObsBasedDiagnosticReportTranslatorImpl.java [hunk 3]
```diff
@@ -79,6 +93,24 @@ private void setPresentedForm(DiagnosticReport diagnosticReport, FhirDiagnosticR
...
+			diagnosticReport.setPerformer(Collections.singletonList(reference));
+		}
+	}
+	
+	private void setPerformer(DiagnosticReport diagnosticReport, FhirDiagnosticReport fhirDiagnosticReport) {
+		if (!diagnosticReport.getPerformer().isEmpty()) {
+			String reference = diagnosticReport.getPerformer().get(0).getReference();
+			String providerUuid = reference.split("/")[1];
+			Provider doctorProvider = providerService.getProviderByUuid(providerUuid);
+			fhirDiagnosticReport.setPerformers(Collections.singleton(doctorProvider));
```
Comment (pajmera4) [inline api/src/main/java/org/openmrs/module/fhirExtension/translators/impl/ObsBasedDiagnosticReportTranslatorImpl.java:110]: I think this logic would be moved away, when practitionerReferenceTranslator toOpenMrsType function is used

File: api/src/test/java/org/openmrs/module/fhirExtension/translators/ObsBasedDiagnosticReportTranslatorTest.java [hunk 4]
```diff
@@ -192,6 +198,54 @@ public void givenFhirDiagnosticReport_WhenOnlyNotesObsArePresent_ShouldNotTransl
...
+		FhirDiagnosticReport fhirDiagnosticReport = new FhirDiagnosticReport();
+		Provider provider = new Provider();
+		Reference reference = new Reference("Practitioner");
+		reference.setReference("Practitioner/123");
+		provider.setUuid("123");
+		provider.setName("Test");
+		fhirDiagnosticReport.setPerformers(singleton(provider));
+		DiagnosticReport diagnosticReport = new DiagnosticReport();
+		when(diagnosticReportTranslator.toFhirResource(fhirDiagnosticReport)).thenReturn(diagnosticReport);
+		when(practitionerReferenceTranslator.toFhirResource(provider)).thenReturn(reference);
```
Comment (pajmera4) [inline api/src/test/java/org/openmrs/module/fhirExtension/translators/ObsBasedDiagnosticReportTranslatorTest.java:212]: We can also put an assertion for reference, as in where this value is been set:
assertEquals("Practitioner/123", result.getPerformers().get(0).getReference())

**Review by pajmera4 — APPROVED** [review]
Looks good!!

File: api/src/main/java/org/openmrs/module/fhirExtension/translators/impl/ObsBasedDiagnosticReportTranslatorImpl.java (hunk 1 above)
Comment (abinaya-u) [reply api/src/main/java/org/openmrs/module/fhirExtension/translators/impl/ObsBasedDiagnosticReportTranslatorImpl.java:47]: Changes are made

File: api/src/main/java/org/openmrs/module/fhirExtension/translators/impl/ObsBasedDiagnosticReportTranslatorImpl.java (hunk 3 above)
Comment (abinaya-u) [reply api/src/main/java/org/openmrs/module/fhirExtension/translators/impl/ObsBasedDiagnosticReportTranslatorImpl.java]: removed the above logic and using practitionerReferenceTranslator toOpenMrsType function

File: api/src/test/java/org/openmrs/module/fhirExtension/translators/ObsBasedDiagnosticReportTranslatorTest.java (hunk 4 above)
Comment (abinaya-u) [reply api/src/test/java/org/openmrs/module/fhirExtension/translators/ObsBasedDiagnosticReportTranslatorTest.java:213]: added this assertion

**Pull request merged**